crates/outbound-http/src/lib.rs

```diff
@@ -49,7 +61,14 @@ impl outbound_http::Host for OutboundHttp {
             }
 
             let method = method_from(req.method);
-            let url = Url::parse(&req.uri).map_err(|_| HttpError::InvalidUrl)?;
+
+            let req_url: Url = if req.uri.starts_with('/') {
```

Comment:
To avoid having the parse and error map on both branches, I wonder if this could be reorganised as something like

```suggestion
            let abs_url = if req.uri.starts_with('/') {
                format!(...)
            } else {
                req.uri.to_string()
            };
            let req_url = Url::parse(&abs_uri).map_err(...)?;
```